controllers/machine_controller_phases.go

```diff
@@ -67,6 +67,12 @@ func (r *MachineReconciler) reconcilePhase(_ context.Context, m *clusterv1.Machi
 		m.Status.SetTypedPhase(clusterv1.MachinePhaseFailed)
 	}
 
+	// Set the phase to "provisioned" if there is a NodeRef field but the infrastructure is not ready
+	// and there are no error messages.
+	if m.Status.NodeRef != nil && !m.Status.InfrastructureReady {
```

Comment:
Can we change line 60 to basically only check if the NodeRef is set, and move to provisioned regardless of Infrastructure readiness?

Reply:
but isn't the status set to `Running` if NodeRef and set and infrastructure is `ready`. 
Also the issue mentions that if the infrastructure vm is restarted, the provider would set the Infrastructure to not ready after it was already set to ready the the machine to running, so need that check to flip it back to provisioned

Reply:
I don't I understand what you mean exactly, can you explain more. Thanks

Reply:
Seems I pointed to the wrong line, it was line 56, not 60.

The line above sets Provisioned when `m.Status.InfrastructureReady` is true. Instead of doing that, we can change it to `if m.Status.NodeRef != nil {`. The Machine will provision to `Running` only if the NodeRef is not nil and the Infrastructure is ready, giving us the behavior we want.

Reply:
oh ok, that makes sense, will change that